### DISCUSSION TITLE
Custom Pinterest Provider Integration: [next-auth][error][GET_AUTHORIZATION_URL_ERROR]

### DISCUSSION THREAD

**Discussion by feliche93** [nextauthjs/next-auth#3204] [Help]
### Question 💬

I am currently working on custom providers for `Notion` and `Pinterest`. 

For the tutorial I followed this [next-auth documentation](https://next-auth.js.org/configuration/providers) and [Pinterest Developer documentation](https://developers.pinterest.com/docs/api/v5/#section/Start-the-OAuth-flow) creating the following object in `[...nextauth].js`

Version: `4.0.0-beta.6`

```js

export default async function auth(req, res) {
    const providers = [
        {
            id: "pinterest",
            name: "Pinterest",
            type: "oauth",
            version: "2.0",
            scope: "boards:read,boards:read_secret,boards:write,boards:write_secret,pins:read,pins:read_secret,pins:write,pins:write_secret,user_accounts:read",
            accessTokenUrl: "https://api.pinterest.com/v5/oauth/token",
            authorizationUrl: "https://www.pinterest.com/oauth&response_type=code",
            requestTokenUrl: "",
            profileUrl: "",
            clientId: process.env.PINTEREST_CLIENT_ID,
            clientSecret: process.env.PINTEREST_CLIENT_SECRET,
            profile: async (profile, accessToken) => {},
            checks: "state",
            headers: {},
            authorizationParams: {
                client_id: process.env.PINTEREST_CLIENT_ID,
                // scope: scope,
                redirect_uri: encodeURIComponent(process.env.PINTEREST_REDIRECT_URI),
            }
        },
    ]
```

This produces the following errors: 
```bash
[next-auth][error][GET_AUTHORIZATION_URL_ERROR] 
https://next-auth.js.org/errors#get_authorization_url_error authorization_endpoint must be configured on the issuer {
  message: 'authorization_endpoint must be configured on the issuer',
  stack: 'TypeError: authorization_endpoint must be configured on the issuer\n' +

[next-auth][error][SIGNIN_OAUTH_ERROR] 
https://next-auth.js.org/errors#signin_oauth_error authorization_endpoint must be configured on the issuer {
  error: {
    message: 'authorization_endpoint must be configured on the issuer',
```

I am pretty new to JS and coming more from a Pyhton background, so might be a beginner mistake. However, when I tried to debug and go deeper in the source code I was a bit list and couldn't understand why the `authorizationUrl` could not be picked up from the object.  Same error happend on Notion, despit having verified there that the authorization url is 100% functional when I copied it to the browser.

Any help or guidance on how to further debug would be very much appreciated, thanks 🙏 

### How to reproduce ☕️

1. Copy object code to `[...nextauth].js`
2. Set env variables: `process.env.PINTEREST_REDIRECT_URI process.env.PINTEREST_CLIENT_ID process.env.PINTEREST_CLIENT_SECRET`
3. Trigger sign in function

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
You are using the v3 provider syntax with `next-auth` v4. Have a look at these:

https://next-auth.js.org/getting-started/upgrade-v4#providers

https://next-auth.js.org/configuration/providers/oauth#options

Looks like we have some outdated docs page on https://next-auth.js.org/configuration/providers

I'll open an issue on it in the docs repo! (https://github.com/nextauthjs/docs/issues/118)


May I ask how did you land on it? Was it from a google search or some navigation?

**Reply by shanoysinc** [reply]
Ran into the same issue was using v3 instead of v4. but it was because of a google search
